src/test/java/org/apache/commons/cli/PatternOptionBuilderTest.java

```diff
@@ -159,13 +161,15 @@ public void testURLPattern() throws Exception
     @Test
     public void testExistingFilePattern() throws Exception
     {
-        final Options options = PatternOptionBuilder.parsePattern("f<");
+        final Options options = PatternOptionBuilder.parsePattern("f<g<");
         final CommandLineParser parser = new PosixParser();
-        final CommandLine line = parser.parse(options, new String[] { "-f", "test.properties" });
+        final CommandLine line = parser.parse(options, new String[] { "-f", "non-existing.file", "-g", "src/test/resources/existing-readable.file" });
+        
+        assertNull("option f parsed", line.getOptionObject("f"));
 
-        assertEquals("f value", new File("test.properties"), line.getOptionObject("f"));
-
-        // todo test if an error is returned if the file doesn't exists (when it's implemented)
+        Object parsedReadableFileStream = line.getOptionObject("g");
+        assertNotNull("option g not parsed", parsedReadableFileStream);
+        assertEquals("option g not FileInputStream", FileInputStream.class, parsedReadableFileStream.getClass());
```

Comment:
How about `assertTrue(parsedReadableFileStream instanceof FileInputStream)`? Your assert will fail for subclasses of `FileInputStream`